services/server/src/lib/fawry.ts

```diff
@@ -0,0 +1,47 @@
+import { GeneralRequestPayload } from "@/fawry/types/requests";
+import { PaymentMethod } from "@/fawry/types/ancillaries";
+import { fawryConfig } from "@/constants";
+import { IUser } from "@litespace/types";
+
+export function forgeFawryPayload<Other = object>({
```

Comment:
I am not a fan of this function for two reasons: 
1. It is using casting. 
2. It accepts the signature as an optional param with default value set to empty string (which is not true and invalid). Also, you relay on the consumer of this function to re-assign the value the signature.